src/compiler/parsing/watf/number-literals.js

```diff
@@ -54,7 +54,7 @@ export function parse32I(sourceString: string): number {
 export function parse64I(sourceString: string): LongNumber {
   let long: Long;
   if (isHexLiteral(sourceString)) {
-    long = Long.fromString(sourceString.substr(2), false, 16);
+    long = Long.fromString(sourceString, false, 16);
```

Comment:
the use of `substr` was removing the sign, and wasn't needed - Long can cope with negative hex values